### PR TITLE
debounce expensive endpoint call

### PR DESCRIPTION
**Pull request by pieterlukasse**
Jira Ticket: [VADC-13](https://ctds-planx.atlassian.net/browse/VADC-13)

### Improvements

- debounce expensive endpoint call that updates the HARE dropdown

### CODE REVIEW

File: src/Analysis/GWASWizard/shared/useDebounce.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          setDebouncedValue(value);
+        }, delay);
+        // Cancel the timeout if value changes (also on delay change or unmount)
+        // This is how we prevent debounced value from updating if value is changed ...
+        // .. within the delay period. Timeout gets cleared and restarted.
+        return () => {
+          clearTimeout(handler);
+        };
+      },
+      [value, delay] // Only re-call effect if value or delay changes
```
Comment (alilahrime) [inline src/Analysis/GWASWizard/shared/useDebounce.js:21]: does this behave the same way when you remove `delay` from dependency array? its always 800 i think

Comment (alilahrime) [reply src/Analysis/GWASWizard/shared/useDebounce.js:21]: run `npm run eslint` if u didnt already

Comment (pieterlukasse) [reply src/Analysis/GWASWizard/shared/useDebounce.js:21]: good point. I'll remove that one.

Comment (pieterlukasse) [reply src/Analysis/GWASWizard/shared/useDebounce.js]: Removing the delay from the useEffect causes this eslint warning though:
```
/Users/pieterlukasse/dev/data-portal/src/Analysis/GWASWizard/shared/useDebounce.js
  23:5  warning  React Hook useEffect has a missing dependency: 'delay'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```
...so decided to keep it.

Comment (alilahrime) [reply src/Analysis/GWASWizard/shared/useDebounce.js]: mkay no worries, thought so

**Review by alilahrime — APPROVED** [review]

**Pull request merged**